code/game/jobs/job_controller.dm

```diff
@@ -464,6 +515,22 @@ var/global/datum/controller/occupations/job_master
 							H.equip_to_slot_or_del(BPK, slot_back,1)
 					H.species.equip(H)
 
+			//Deferred item spawning.
+			if(spawn_in_storage && spawn_in_storage.len)
+				var/obj/item/weapon/storage/B
+				for(var/obj/item/weapon/storage/S in H.contents)
+					B = S
+					break

```

Comment:
I assume you were looking at Polaris' implementation?
It's a bit dated compared to Bay's that, among other things, first checks that the item actually fits in the found storage (and if that fails tries to put the item in hand, etc.).